LayoutTests/inspector/console/js-completions.html

```diff
@@ -52,6 +52,50 @@
         }
     });
 
+    suite.addTestCase({
+        name: "console.jsCompletions.completionOrdering",
+        description: "Test that the JavaScript completions order its items in the expected way.",
+        async test() {
+            const verifyCompletionsOrdering = async (prefix) => {
+                let completions = await generateJSCompletions({prefix});
+                for (let i = 0; i < completions.length - 1; ++i) {
+                    let s = completions[i];
+                    let t = completions[i + 1];
+
+                    let sIsNumerical = isFinite(s);
+                    let tIsNumerical = isFinite(t);
+                    if (sIsNumerical && tIsNumerical) {
+                        InspectorTest.assert(+s <= +t, `Completions should order numerical items with lower values to come first. Got "${s}" coming before "${t}".`);
+                        continue;
+                    }
+                    if (sIsNumerical != tIsNumerical) {
+                        InspectorTest.assert(sIsNumerical, `Completions should order numerical items to come before non-numerical items. Got "${s}" coming before "${t}".`);
+                        continue;
+                    }
+
+                    let sIsRareProperty = s.startsWith("__") && s.endsWith("__");
+                    let tIsRareProperty = t.startsWith("__") && t.endsWith("__");
+                    if (sIsRareProperty != tIsRareProperty) {
+                        InspectorTest.assert(tIsRareProperty, `Completions should order non-rare properties before rare properties. Got "${s}" coming before "${t}".`);
+                        continue;
+                    }
+
+                    let sMatchesPrefix = s.startsWith(prefix);
+                    let tMatchesPrefix = t.startsWith(prefix);
+                    if (sMatchesPrefix != tMatchesPrefix) {
+                        InspectorTest.assert(sMatchesPrefix, `Completions should order items case-sensitively matching the given prefix ("${prefix}") first. Got "${s}" coming before "${t}".`);
+                        continue;
+                    }
+
+                    InspectorTest.assert(s.extendedLocaleCompare(t) <= 0, `Completions should order items according to extendedLocaleCompare. Got "${s}" coming before "${t}".`);
+                }
+            };
+
+            await verifyCompletionsOrdering("");
+            await verifyCompletionsOrdering("doc");
```

Comment:
can we try a few more?
```js
await verifyCompletionsOrdering("__pro");
await verifyCompletionsOrdering("1");
```
and maybe add a `window["123"] = "foo"` as well as a `window["__pro_test"] = "normal"` and `window["__pro_test__"] = "rare"`?